examples/proc_GDS_capture_int.py

```diff
@@ -0,0 +1,108 @@
+import pyspecdata as psd
+import matplotlib.pyplot as plt
+import numpy as np
+from itertools import cycle
+
+colorcyc_list = plt.rcParams["axes.prop_cycle"].by_key()["color"]
+color_cycle = cycle(colorcyc_list)
+
+V_atten_ratio = 102.35  # attenutation ratio
+loose_HH_width = 5e6
+int_slop = 1e-6
+
+with psd.figlist_var() as fl:
+    for filename, nodename in [
+        (
+            "240812_amp0p1_deblank_50_pulse_capture.h5",
+            "pulse_capture_1",
+        ),
+    ]:
+        s = psd.find_file(
+            filename, expno=nodename, exp_type="ODNP_NMR_comp/test_equipment"
+        )
+        amplitude = s.get_prop("acq_params")["amplitude"]
+        beta_us_sqrt_W = s.get_prop("acq_params")["beta_90_s_sqrtW"]
+        fl.basename = (
+            f"amplitude = {amplitude}, $\\beta$ = {beta_us_sqrt_W} \n"
+        )
+        if not s.get_units("t") == "s":
+            print(
+                "units weren't set for the t axis or else I can't read them from the hdf5 file!"
+            )
+            s.set_units("t", "s")
+        s *= V_atten_ratio  # attenutation ratio
+        s /= np.sqrt(50)  # V/sqrt(R) = sqrt(P)
+        fl.next(r"$\sqrt{P_{analytic}}$ vs $t_{pulse}$")
+        raw_color = next(color_cycle)
+        fl.plot(s, color=raw_color, label="raw analytic")
+        abs_color = next(color_cycle)
+        fl.plot(abs(s), color=abs_color, label="abs(analytic)")
+        # {{{ apply frequency filter
+        SW = 1 / (s["t"][1] - s["t"][0])  # sample rate
+        carrier = (
+            s.get_prop("acq_params")["carrierFreq_MHz"] * 1e6
+        )  # signal frequency
+        fn = SW / 2  # nyquist frequency
+        n = np.floor(
+            (carrier + fn) / SW
+        )  # how far is the carrier from the left side of the spectrum (which is at SW/2), in integral multiples of SW
+        nu_a = (
+            carrier - n * SW
+        )  # find the aliased peak -- again, measuring from the left side
+        center = SW - abs(nu_a)
+        s.ft("t")
+        fl.next("Frequency Domain")
+        fl.plot(s, color=raw_color, label="raw analytic")
+        fl.plot(abs(s), color=abs_color, label="abs(analytic)")
+        # {{{ lorentzian filter
+        x = s.C.getaxis("t")
+        for_L = s.C  # make copy for lorentzian filter
+        Lambda = 15.19e6 - 14.61e6
+        s_max = abs(for_L).data.max()
+        L = (
+            psd.nddata(
+                s_max / (1 + 1j * 2 * (x - carrier) * (1 / Lambda)), ["t"]
+            )
+            .setaxis("t", x)
+            .set_units("t", for_L.get_units("t"))
+        )
+        for_L *= L
+        for_L /= s_max
+        # }}}
+        # {{{ heaviside hat functions
+        # {{{ determine center frequency
+        s.ift("t")
+        # }}}
+        s.ft("t")
+        wide_HH = s.C
+        wide_HH["t" : (0, center - loose_HH_width)] *= 0
```

Comment:
do not use this "center" thing.  Use nu_a.  It might be a negative number, and that's OK -- that will happen if you alias your signal to negative.

Also, add a line here -- if nu_a is negative, check that plus or minus HH width doesn't include zero.  Also, you need to multiply HH width by 0.5 here, or else it's not a width.

Reply:
for the "add a line" I'm just saying
```
assert (
             0 > nu_a + 0.5*HH_width ) or ( 0 < nu_a - 0.5*HH_width), "unfortunately the region I want to filter includes DC -- this is probably not good, and means you should pick a different timescale for your scope so this doesn't happen"
             ```

Reply:
https://jmfrancklab.slack.com/archives/CV553TPRV/p1724163442806419

Reply:
we discussed this -- sounds like either comments or change are in progress